client/components/navigation-header/navigation-header.tsx

```diff
@@ -50,21 +66,23 @@ const NavigationHeader: React.FC< HeaderProps > = ( {
 			← { backLinkProps?.text ?? translate( 'Back' ) }
 		</a>
 	),
-	children,
+	rightSection,
 	hasScreenOptionsTab,
 	...rest
 } ) => {
 	return (
 		<header
-			className={ clsx( 'calypso-navigation-header', className, {
+			className={ clsx( 'calypso-navigation-header', 'navigation-header', className, {
 				'calypso-navigation-header__screen-options-tab': hasScreenOptionsTab,
 			} ) }
 			{ ...rest }
 		>
 			<div className="calypso-navigation-header__head">{ headElement }</div>
 			<div className="calypso-navigation-header__body">
 				<div className="calypso-navigation-header__left-section">{ titleElement }</div>
-				<div className="calypso-navigation-header__right-section">{ children }</div>
+				{ rightSection && (
```

Comment:
Is introducing this `rightSection` a kind of [YAP-ing](https://dio.la/article/the-everything-bagel-of-components#yap-ing) mentioned by @kangzj in the task `STATS-6` of creating the component?

Reply:
The `rightSection` still takes in the whole component as to what is to be displayed in that section. IMO, the wrapper for the right section is required in the component itself, as it acts as a slot and places the content in the correct location. Let me know your thoughts.

Reply:
Per the original component design, I feel like the `children` property was introduced to represent the `right section` here. 🤔 @kangzj What do you think?

Reply:
`rightSection` feels like a general name to me and I don't think it's YAPPing. It doesn't have to be `children` because the component has several children (slots) and no primary one. Thoughts?